packages/store-ui/src/molecules/Modal/Modal.tsx

```diff
@@ -0,0 +1,116 @@
+import type {
+  AriaAttributes,
+  KeyboardEvent,
+  MouseEvent,
+  PropsWithChildren,
+} from 'react'
+import React from 'react'
+import { createPortal } from 'react-dom'
+
+import type { OverlayProps } from '../../atoms/Overlay'
+import Overlay from '../../atoms/Overlay'
+import ModalContent from './ModalContent'
+import type { ModalContentProps } from './ModalContent'
+
+const composeEventHandler = <EventType extends React.SyntheticEvent | Event>(
+  internalHandler: (event: EventType) => void
+) => {
+  return (event: EventType) => {
+    if (!event.defaultPrevented) {
+      internalHandler(event)
+    }
+  }
+}
```

Comment:
I don't think this function aggregates a lot to the code, I think it makes the functions that it wraps hard to read. I'd remove it and put the code directly on the callbacks, even if it means repeating a little bit of code.

Reply:
Yeah, i totally agree with you.